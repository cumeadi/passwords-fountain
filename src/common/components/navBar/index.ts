export { NavBar } from './navBar.component';
