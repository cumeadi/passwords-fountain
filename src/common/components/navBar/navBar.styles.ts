import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme, typography, media, stylelint } from '@/common/theme';
setPragma(h);

export const Wrapper = styled('nav')`
    display: grid;
    grid-template-columns: 1fr;
    grid-template-areas:
        'logo . appLink'
        'links links links';

    ${media.gte(theme.breakpoints.m920)(stylelint.css`
        grid-template-areas: 'logo links appLink';
    `)}
`;

export const LogoWrapper = styled('div')`
    grid-area: logo;
    display: flex;
`;

export const LogoAnchor = styled('a')`
    text-decoration: none;
    display: grid;
    grid-template-columns: auto 1fr;
    grid-gap: ${theme.spacing.s12};
    align-items: center;
    padding: ${theme.spacing.m18} 0;
`;

export const AppName = styled('h1')``;

export const AppLinkWrapper = styled('div')`
    grid-area: appLink;
    padding: ${theme.spacing.m18} 0;
    display: grid;
    place-items: center;
`;

const PrimaryText = stylelint.css`
    font-family: ${theme.fontFamilies.bold};
    
    ${media.gte(theme.breakpoints.s375)(stylelint.css`
        ${typography.text24}
    `)}
    
    ${typography.text20}
`;

export const AppLinkAnchor = styled('a')`
    display: flex;
    align-items: center;
    text-decoration: none;
    font-family: ${theme.fontFamilies.bold};
    color: ${theme.colors.darkBlue};

    &:hover {
        text-decoration: underline;
    }

    ${PrimaryText}
`;

export const Green = styled('span')`
    color: ${theme.colors.primaryGreen};
    margin-right: ${theme.spacing.xs6};
    ${PrimaryText}
`;

export const Blue = styled('span')`
    color: ${theme.colors.primaryBlue};
    ${PrimaryText}
`;

const LinksPadding = `${theme.spacing.m18} 0 ${theme.spacing.m18}`;
export const LinksWrapper = styled('div')`
    grid-area: links;
    display: grid;
    grid-auto-flow: column;
    justify-items: center;
    align-items: center;
    padding: ${({ withoutPadding }: { withoutPadding: boolean }) =>
        withoutPadding ? 0 : LinksPadding};

    ${media.gte(theme.breakpoints.m920)(stylelint.css`
        justify-content: flex-end;
    `)}
`;

export const LinkAnchor = styled('a')`
    color: ${theme.colors.darkBlue};
    text-decoration: none;

    &:hover {
        text-decoration: underline;
    }

    ${media.gte(theme.breakpoints.s375)(stylelint.css`
        ${typography.text20}
    `)}
    
    ${media.gte(theme.breakpoints.m700)(stylelint.css`
        ${typography.text24}
    `)}
    
    ${media.gte(theme.breakpoints.m920)(stylelint.css`
        margin-right: ${theme.spacing.xxl60};
    `)}
     
    ${typography.text18}
`;
