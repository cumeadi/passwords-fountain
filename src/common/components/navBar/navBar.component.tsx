import { h, VNode } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import { Link, useLocation } from 'wouter-preact';
import {
    Wrapper,
    LogoWrapper,
    AppName,
    Green,
    Blue,
    LinksWrapper,
    LinkAnchor,
    AppLinkWrapper,
    AppLinkAnchor,
    LogoAnchor,
} from './navBar.styles';
import { Icon } from '@/common/components/icon';
import { Text } from '@/modules/localisation/localisation.context';
import { renderIfTrue } from '@/common/utils/rendering';

interface NavLink {
    id: number;
    label: string;
    href: string;
}

const navLinks: NavLink[] = [
    {
        id: 0,
        label: 'navBar.howToStart',
        href: '#how-to-start',
    },
    {
        id: 1,
        label: 'navBar.pricing',
        href: '#pricing',
    },
    {
        id: 2,
        label: 'navBar.authors',
        href: '#authors',
    },
];

export const NavBar: TypedComponent<Props> = () => {
    const logoIconSize = 40;
    const [location] = useLocation();
    const isOnHomepage = location.length === 1;
    const appLink: NavLink = {
        id: 3,
        label: 'navBar.app',
        href: isOnHomepage ? '/app' : '/',
    };

    const renderNavLinks = renderIfTrue((): VNode[] => {
        return navLinks.map(navLink => (
            <LinkAnchor key={navLink.id} href={navLink.href}>
                <Text>{navLink.label}</Text>
            </LinkAnchor>
        ));
    });

    const renderAppLinkLabel = () => {
        return isOnHomepage ? (
            <Text>{appLink.label}</Text>
        ) : (
            <Icon name="home" width={32} height={32} />
        );
    };

    return (
        <Wrapper>
            <LogoWrapper>
                <Link href="/">
                    <LogoAnchor>
                        <Icon
                            name="logo"
                            width={logoIconSize}
                            height={logoIconSize}
                        />
                        <AppName>
                            <Green>Passwords</Green>
                            <Blue>Fountain</Blue>
                        </AppName>
                    </LogoAnchor>
                </Link>
            </LogoWrapper>
            <LinksWrapper withoutPadding={!isOnHomepage}>
                {renderNavLinks(isOnHomepage)}
            </LinksWrapper>
            <AppLinkWrapper>
                <Link href={appLink.href}>
                    <AppLinkAnchor>{renderAppLinkLabel()}</AppLinkAnchor>
                </Link>
            </AppLinkWrapper>
        </Wrapper>
    );
};

interface Props {}

NavBar.propTypes = {};
