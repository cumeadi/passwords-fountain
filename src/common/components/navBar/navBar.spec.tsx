import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { NavBar } from './navBar.component';

describe('NavBar', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<NavBar />);

        expect(asFragment()).toMatchSnapshot();
    });
});
