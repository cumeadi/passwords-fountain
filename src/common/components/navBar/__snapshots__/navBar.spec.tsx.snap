// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="go956739197"
  >
    <div
      class="go1351599409"
    >
      <a
        class="go2191268173"
        href="/"
      >
        <img
          alt="logo icon"
          class="go115679"
          height="40"
          src="static-asset-stub"
          width="40"
        />
        <h1
          class="go115679"
        >
          <span
            class="go2484009180"
          >
            Passwords
          </span>
          <span
            class="go792706930"
          >
            Fountain
          </span>
        </h1>
      </a>
    </div>
    <div
      class="go1198697951"
    >
      <a
        class="go1602260"
        href="#how-it-works"
        native="true"
      >
        How it works
      </a>
      <a
        class="go1602260"
        href="#how-to-start"
        native="true"
      >
        How to start
      </a>
      <a
        class="go1602260"
        href="#pricing"
        native="true"
      >
        Pricing
      </a>
    </div>
    <div
      class="go123319007"
    >
      <section
        class="go2164034101"
      >
        <section
          class="go62670245"
          tabindex="0"
        >
          <div
            class="go3942427238"
          >
            <img
              alt="globe icon"
              class="go115679"
              height="30"
              src="static-asset-stub"
              width="30"
            />
          </div>
          <div
            class="go129576329"
          />
        </section>
      </section>
    </div>
  </nav>
</DocumentFragment>
`;
